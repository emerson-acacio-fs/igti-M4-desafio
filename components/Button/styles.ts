import { ButtonHTMLAttributes } from "react"
import styled, { css } from "styled-components"

export const ButtonWrapper = styled.button`
  ${({ theme }) => css`
    padding: 1rem;
    background-color: ${theme.colors.white};
    border: 0.1rem solid ${theme.colors.black};
    border-radius: ${theme.border.radius};
    font-size: ${theme.font.sizes.medium};
    cursor: pointer;
    &:disabled {
      cursor: not-allowed;
    }
  `}
`
