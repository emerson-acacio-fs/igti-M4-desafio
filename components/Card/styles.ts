import { ButtonHTMLAttributes } from "react"
import styled, { css } from "styled-components"

export const CardWrapper = styled.li`
  ${({ theme }) => css`
    width: 100%;
    border-radius: 1rem;
    text-decoration: none;
    list-style: none;
    background-color: ${theme.colors.lightGray};
    display: flex;
    padding: 1.5rem;
    column-gap: 1.5rem;
  `}
`

export const InfoWrapper = styled.div``

export const ImgWrapper = styled.div`
  ${({ theme }) =>
    css`
      width: 5rem;
      height: 5rem;
      img {
        width: 100%;
        height: 100%;
        object-fit: cover;
      }
    `}
`
