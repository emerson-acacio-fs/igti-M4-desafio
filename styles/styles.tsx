import styled, { css } from "styled-components"

export const ListWrapper = styled.ul`
  li {
    margin-bottom: 1.5rem;
  }
`
export const HeaderWrapper = styled.ul`
  display: flex;
  justify-content: space-between;
`

export const Main = styled.main`
  margin: 1.5rem auto;
  min-width: 740px;
  max-width: 1180px;
  display: flex;
  flex-direction: column;
  row-gap: 1.5rem;
`
