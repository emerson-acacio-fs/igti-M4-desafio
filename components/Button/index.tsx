import { ButtonHTMLAttributes, ReactNode } from "react"
import * as S from "./styles"

export function Button({
  children,
  disabled,
  ...props
}: ButtonHTMLAttributes<HTMLButtonElement>): JSX.Element {
  return (
    <S.ButtonWrapper disabled={disabled} {...props}>
      {!!children && children}
    </S.ButtonWrapper>
  )
}
