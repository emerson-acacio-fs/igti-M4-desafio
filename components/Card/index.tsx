import Image from "next/image"
import * as S from "./styles"
export interface ICardProps {
  id: string
  name: string
  year_established: number
  country: string
  image: string
  trust_score: number
  trade_volume_24h_btc: number
}
export function Card({
  name,
  image,
  trust_score,
  year_established,
}: ICardProps): JSX.Element {
  return (
    <S.CardWrapper>
      <S.ImgWrapper>
        {image !== "missing_small.png" && (
          <Image src={image} alt={name} width={50} height={50} />
        )}
      </S.ImgWrapper>
      <S.InfoWrapper>
        <h1>{name}</h1>
        <div>{trust_score}</div> <div>{year_established}</div>
      </S.InfoWrapper>
    </S.CardWrapper>
  )
}
