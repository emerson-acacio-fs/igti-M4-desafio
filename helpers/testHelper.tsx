import { ReactElement } from "react"
import { ThemeProvider } from "styled-components"
import { theme } from "styles/theme"
import { render, RenderOptions } from "@testing-library/react"
import renderer, { TestRendererOptions } from "react-test-renderer"
const customRender = (ui: ReactElement, renderOptions: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions)

const toJSON = (ui: ReactElement, renderOptions?: TestRendererOptions) =>
  renderer
    .create(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions)
    .toJSON()

export { customRender as render, toJSON }
