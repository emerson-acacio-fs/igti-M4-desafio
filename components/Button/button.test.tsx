import { screen } from "@testing-library/react"
import { render, toJSON } from "helpers/testHelper"
import { Button } from "components/Button"
import { theme } from "styles/theme"

describe("Button", () => {
  it("renders a button", () => {
    render(<Button>Página Anterior</Button>)

    const button = screen.getByRole("button", {
      name: /Página Anterior/i,
    })

    expect(button).toBeInTheDocument()
  })

  it("renders with background color white", () => {
    render(<Button>Página Anterior</Button>)

    const button = screen.getByRole("button", {
      name: /Página Anterior/i,
    })

    expect(button).toHaveStyle({
      backgroundColor: theme.colors.white,
      border: `0.1rem solid ${theme.colors.black}`,
    })
  })
  it("should render a disabled input", () => {
    const tree = toJSON(<Button disabled>Página Anterior</Button>)

    expect(tree).toHaveStyleRule("cursor", "not-allowed", {
      modifier: ":disabled",
    })
    expect(tree).toMatchSnapshot()
  })
})
