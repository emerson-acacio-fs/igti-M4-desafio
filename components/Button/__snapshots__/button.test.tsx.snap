// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a disabled input 1`] = `
.c0 {
  padding: 1rem;
  background-color: #FAFAFA;
  border: 0.1rem solid #030517;
  border-radius: 0.4rem;
  font-size: 1.6rem;
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

<button
  className="c0"
  disabled={true}
>
  PÃ¡gina Anterior
</button>
`;
